prometheus/promhttp/http.go

```diff
@@ -133,10 +124,9 @@ func HandlerFor(reg prometheus.Gatherer, opts HandlerOpts) http.Handler {
 		}
 
 		contentType := expfmt.Negotiate(req.Header)
-		buf := getBuf()
-		defer giveBuf(buf)
-		writer, encoding := decorateWriter(req, buf, opts.DisableCompression)
-		enc := expfmt.NewEncoder(writer, contentType)
+		buf := &bytes.Buffer{}
```

Comment:
Why did you remove the sync.Pool for buf? This will create a lot of allocations in high-scrape-rate scenarios.

Reply:
Or to put it another way: For the same reason we want the gzip writer in a sync.Pool, we want this intermediate buffer in a sync.Pool.